JM/CF/Scripts/5_Mission/CommunityFramework/Mission/MissionBase.c

```diff
@@ -23,6 +23,17 @@ modded class MissionBase
 
 	void CF_OnUpdate(float timeslice)
 	{
+#ifdef SERVER
```

Comment:
How is timeslice "inaccurate"? It's supposed to be used that way, it is used in all of the game's code. Besides floating arithmetic errors that can happen from time to time, I don't see any reason why to make additional function calls.

Also, this code does not anticipate lag spikes, it needs to accumulate updates and then
```
while(accumulator >= 0.025)
{
    DoSomething();
    accumulator -= 0.025;
}
```

So that any update logic behind it can rely on a max update rate of 0.025 for simulation constraints.

Reply:
Yes, you accumulate but only flush once with delta times that can be bigger than 0.025. I thought the intention was a fixed simulation timestamp. If you wanted to limit the max refresh rate only I guess that works.

> timeslice not being accurate in the DayZGame update loop specifically is a DayZ bug.

In what way? What bug? If there would be a bug the entire game would be affected by it. If so it would be patched in the base game anyway, so no need to waste resources in CF code to bypass an intended information source. Give steps to reproduce on the fb tracker. For CF code please revert back to simply using the delta time that is given by the game.

Reply:
> I thought the intention was a fixed simulation timestamp. If you wanted to limit the max refresh rate only I guess that works.

Yes. It will be fixed in the sense that when a CF OnUpdate loop finishes within the allotted time, it will not go above 25 ms. There needs to be allowance for that though, as time taken for one loop can never be guaranteed.

> If there would be a bug the entire game would be affected by it. If so it would be patched in the base game anyway, so no need to waste resources in CF code to bypass an intended information source.

I don't think this particular issue is fixable in the base game due to rounding and floating point precision limits. It doesn't really matter for the CF side anyway. The GetGame().GetTickTime() call doesn't add any measurable overhead (GameLabs/CF Tools do the same to facilitate measuring OnUpdate tick rate, which would not be accurate if they were to rely on the timeslice).